File: Exercicios/Labs/Peso.java
import java.util.Scanner;

class Atleta {
    String nome; 
    int peso;   

    Atleta(String nome, int peso) { // construtor Atleta
        this.nome = nome;          
        this.peso = peso;           
    }
}

public class Peso {

    public static void main(String[] args) {

        Scanner teclado = new Scanner(System.in); 

        Atleta[] atletas = new Atleta[100]; // vetor para armazenar no max 100 atletas

        int n = 0; 

        while (teclado.hasNext()) { 

            String nome = teclado.next(); // le o nome 

            int peso = teclado.nextInt(); // le o peso max

            atletas[n] = new Atleta(nome, peso); // cria um atleta e guarda no vetor

            n++; // ++ quantidade de atletas lidos
        }

        for (int i = 0; i < n - 1; i++) { 

            for (int j = 0; j < n - 1 - i; j++) { 

                boolean trocar = false; // boolean: atletas devem trocar de posição

                // se o peso atual for menor que o prox, troca para deixar o maior peso antes
                if (atletas[j].peso < atletas[j + 1].peso) { 
                    trocar = true;    
                } else if (atletas[j].peso == atletas[j + 1].peso) { // Se os pesos forem iguais

                    if (atletas[j].nome.compareTo(atletas[j + 1].nome) > 0) { // se o nome atual vem depois alfabeticamente
                        trocar = true; // troca para colocar em ordem alfabetica
                    }
                }

                if (trocar) { // se precisar trocar

                    Atleta temp = atletas[j]; // guarda o atleta em temp

                    atletas[j] = atletas[j + 1]; // coloca o prox atleta em atual

                    atletas[j + 1] = temp; // coloca o atleta salvo em prox
                }
            }
        }

        for (int i = 0; i < n; i++) { // percorre e imprime

            System.out.println(atletas[i].nome + " " + atletas[i].peso); 
        }

        teclado.close(); 
    }
}

File: .vscode/java-bin/peso.c
#include <stdio.h>   
#include <string.h>  

typedef struct {
    char nome[51]; 
    int peso;      
} Atleta;

int main() {
    Atleta atletas[100]; // max 100 

    int n = 0; 

    // le até o fim do arquivo
    while (scanf("%s %d", atletas[n].nome, &atletas[n].peso) != EOF) {
        n++; 
    }

    // usando bubble sort
    for (int i = 0; i < n - 1; i++) {
        // percorre comparando pares vizinhos
        for (int j = 0; j < n - 1 - i; j++) {
            int trocar = 0;

            // se o atleta atual levantou menos peso que prox, troca
            if (atletas[j].peso < atletas[j + 1].peso) {
                trocar = 1;
            }

            // se os pesos forem iguais, compara os nomes em ordem alfabetica
            else if (atletas[j].peso == atletas[j + 1].peso) {
                
                // strcmp retorna valor positivo se o primeiro nome vem depois do segundo
                if (strcmp(atletas[j].nome, atletas[j + 1].nome) > 0) {
                    trocar = 1;
                }
            }

            // se precisar trocar
            if (trocar == 1) {
                Atleta temp = atletas[j];  // guarda o atual em temp

                atletas[j] = atletas[j + 1];  // coloca o proximo na atual

                atletas[j + 1] = temp;   // Coloca na proxima posição
            }
        }
    }

    // percorre e imprime os ordenados 
    for (int i = 0; i < n; i++) {
        printf("%s %d\n", atletas[i].nome, atletas[i].peso);
    }

    return 0;
}